aws/ec2_test.go

```diff
@@ -222,3 +243,438 @@ func TestNukeInstances(t *testing.T) {
 		assert.NotContains(t, instances, *instanceID)
 	}
 }
+
```

Comment:
Since these mock based tests are completely different from any other test in cloud-nuke, I think they should be extracted into their own file (e.g `ec2_unit_test.go`) with a comment at the top to help guide someone with the nature of the test. It took me a bit to understand what was going on in these tests (setup mocks to expect a bunch of calls => run the function)